### ISSUE TITLE
[Feature request] Smarter shadow windows

### ISSUE THREAD

**Issue by narcolepticinsomniac** [schomery/popup-blocker#73]
I think this is an underutilized feature in general. It could probably be integrated with a little logic to kick in on its own if it detects that a site tries to open a popup which isn't accepted and then tries to open subsequent popups.

 In these scenarios it's obvious they're malicious, and the biggest problem never solved is recursive loops, [for example](https://123movieshub.to/), which sabotage functionality. Shadow windows were a great idea, but they need to kick in automatically in obvious scenarios to really be put to their best use.

**Comment by schomery** [comment]
Can we have an updated link please.

**Comment by narcolepticinsomniac** [comment]
Yeah, I think that site went down like a week after posting the link. They've been dropping like flies the past year. I don't currently have a good example of the recursive loop issue, but utilizing shadow windows automatically seems like a no-brainer to me regardless. If a site tries to open the same popups after one isn't accepted, banish them to the Shadow DOM..
